pkg/core/serving/kedahttp/servingrun.go

```diff
@@ -0,0 +1,585 @@
+			GenerateName: fmt.Sprintf("%s-service-%s-", s.Name, version),
+			Namespace:    s.Namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports:    []corev1.ServicePort{svcPort},
+			Selector: labels,
+		},
+	}
+
+	return service, nil
+}
+
+func (r *servingRun) createScaler(s *openfunction.Serving, workload runtime.Object, service *corev1.Service) error {
+	log := r.log.WithName("CreateKedaScaler").
+		WithValues("Serving", fmt.Sprintf("%s/%s", s.Namespace, s.Name))
+
+	// When no Triggers are configured, it means that no scaler needs to be created for the function.
+	if s.Spec.ScaleOptions == nil || s.Spec.ScaleOptions.Keda == nil {
+		log.Info("No keda scaleOptions found, no need to create scaler.")
+		return nil
+	}
+
+	version := constants.DefaultFunctionVersion
+	if s.Spec.Version != nil {
+		version = *s.Spec.Version
+	}
+	version = strings.ReplaceAll(version, ".", "")
+
+	keda := s.Spec.ScaleOptions.Keda
+
+	var hosts []string
+	for _, hostname := range s.Spec.Triggers.Http.Route.Hostnames {
+		hosts = append(hosts, string(hostname))
+	}
+
+	var pathPrefix []string
+	for _, rule := range s.Spec.Triggers.Http.Route.Rules {
+		for _, match := range rule.Matches {
+			if *match.Path.Value == "" {
+				pathPrefix = append(pathPrefix, "/")
+			} else {
+				pathPrefix = append(pathPrefix, *match.Path.Value)
+			}
+		}
+	}
+
+	var targetPendingRequests int32 = 100
+	if keda.HTTPScaledObject.TargetPendingRequests != nil {
+		targetPendingRequests = *keda.HTTPScaledObject.TargetPendingRequests
+	}
+	var cooldownPeriod int32 = 300
+	if keda.HTTPScaledObject.CooldownPeriod != nil {
+		cooldownPeriod = *keda.HTTPScaledObject.CooldownPeriod
+	}
+
+	accessor, _ := meta.Accessor(workload)
+
+	httpScaledObject := &httpv1alpha1.HTTPScaledObject{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-scaler-", s.Name),
+			Namespace:    s.Namespace,
+			Labels: map[string]string{
+				common.OpenfunctionManaged: "true",
+				common.ServingLabel:        s.Name,
+			},
+		},
+		Spec: httpv1alpha1.HTTPScaledObjectSpec{
+			Hosts:        hosts,
+			PathPrefixes: pathPrefix,
+			ScaleTargetRef: httpv1alpha1.ScaleTargetRef{
+				Deployment: accessor.GetName(),
+				Service:    service.GetName(),
+				Port:       80,
+			},
+			Replicas: &httpv1alpha1.ReplicaStruct{
+				Min: s.Spec.ScaleOptions.MinReplicas,
+				Max: s.Spec.ScaleOptions.MaxReplicas,
+			},
+			TargetPendingRequests: &targetPendingRequests,
+			CooldownPeriod:        &cooldownPeriod,
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(s, httpScaledObject, r.scheme); err != nil {
+		log.Error(err, "Failed to SetControllerReference")
+		return err
+	}
+
+	if err := r.Create(r.ctx, httpScaledObject); err != nil {
+		log.Error(err, "Failed to create Keda scaler")
+		return err
+	}
+
+	s.Status.ResourceRef[scalerName] = httpScaledObject.GetName()
+
+	log.V(1).Info("Keda scaler Created", "Scaler", httpScaledObject.GetName())
+
+	var ingressClassName = "nginx"
```

Comment:
For now, keda http-addon v0.5.0 seems like only supporting the nginx-ingress.
https://github.com/kedacore/http-add-on/issues/33
https://github.com/kedacore/http-add-on/issues/350
Two issues above mentioned this problem brefore, there may be more choices in the future. So I'm using this just for now, and I'll add some comments in this part of code, so that maybe oneday, this can be specified by the user.

Reply:
We already have added K8s Gateway API support,  and still need to create an ingress backed by nginx for `keda-http-addon`? For this addon to work, we still need to install nginx? It doesn't sound reasonable.